src/main/plugin/iso19139.ca.HNAP/update-fixed-info.xsl

```diff
@@ -876,9 +876,18 @@
 
 
             <gmd:type>
-              <gmd:MD_KeywordTypeCode codeList="https://schemas.metadata.geo.ca/register/napMetadataRegister.xml#IC_101" codeListValue="{@type}">
-                  <xsl:value-of select="$codelistDocument/codelists/codelist[@name='gmd:MD_KeywordTypeCode']/entry[code = $currentCodeValue]/value"/>
-              </gmd:MD_KeywordTypeCode>
+              <xsl:choose>
+                <xsl:when test="$currentCodeValue='theme'">
+                  <gmd:MD_KeywordTypeCode codeList="https://schemas.metadata.geo.ca/register/napMetadataRegister.xml#IC_101" codeListValue="RI_528">
+                    <xsl:value-of select="$codelistDocument/codelists/codelist[@name='gmd:MD_KeywordTypeCode']/entry[code = 'RI_528']/value"/>
+                  </gmd:MD_KeywordTypeCode>
+                </xsl:when>
+                <xsl:otherwise>
+                  <gmd:MD_KeywordTypeCode codeList="https://schemas.metadata.geo.ca/register/napMetadataRegister.xml#IC_101" codeListValue="{@type}">
```

Comment:
Have you check in this case, what contains `{@type}`?

Reply:
The type is the codelist itself for example RI_528. This code is direct copy/paste from the original code https://github.com/metadata101/iso19139.ca.HNAP/blob/43efe958be6b8eb18acec2372e945056037b18e2/src/main/plugin/iso19139.ca.HNAP/update-fixed-info.xsl#L879

The only logic change is is to determine if the codelist is hardcoded to "theme" and put RI_528. Rest of the code are untouched.